staging/src/k8s.io/apimachinery/pkg/runtime/error.go

```diff
@@ -120,3 +120,32 @@ func IsMissingVersion(err error) bool {
 	_, ok := err.(*missingVersionErr)
 	return ok
 }
+
+// strictnessError is a base error type that is returned by a strict Decoder such
+// as UniversalStrictDecoder.
+type strictnessError struct {
+	message string
+	data    string
+}
+
+// NewStrictDecoderError creates a new strictnessError object
+func NewStrictnessError(message string, data string) error {
```

Comment:
prefer `NewStrictDecodingError`, `type strictDecodingError struct`, and `IsStrictDecodingError`

Reply:
also please make sure the error type matches in the comments about function names e.g. here:
https://github.com/kubernetes/kubernetes/pull/74111/files#diff-f216f544515d2fd05d66d92c5f95a248R49